app/actors/hyku_addons/actors/json_fields_actor.rb

```diff
@@ -14,18 +14,22 @@ def update(env)
 
         def jsonify_fields(env)
           env.curation_concern.class.json_fields.each do |field|
+            # This handles the case when field is a key/value pair coming from the yaml schema
+            field = field.first if field.is_a?(Array)
+
             if name_blank?(field, env.attributes[field]) || recursive_blank?(env.attributes[field])
               env.attributes.delete(field)
             else
-              env.attributes[field].reject! { |o| name_blank?(field, o) || recursive_blank?(o) } if env.attributes[field].is_a? Array
+              env.attributes[field].reject! { |o| name_blank?(field, o) || recursive_blank?(o) } if env.attributes[field].is_a?(Array)
```

Comment:
You define `name_blank?` and `recursive_blank?` identically in two places.  Also when it's used it's always `name_blank? || recursive_blank?`.  Could you move both definitions to a module, and create one method which will test the full condition?

Reply:
This is part of the original json actor that was inherited from the OG PR. I have tried not to touch as much as possible, but you're probably right that extra work might want to be done to simplify it here. I would suggest its best to leave that for now, as we know it works currently and so it could be discounted if errors start to pop up.